osu.Game/Screens/Select/SongSelect.cs

```diff
@@ -619,12 +618,30 @@ public override bool OnExiting(IScreen next)
 
             FilterControl.Deactivate();
 
-            music.CurrentTrack.Looping = false;
-            music.TrackChanged -= ensureTrackLooping;
+            endLooping();
 
             return false;
         }
 
+        private bool isHandlingLooping;
+
+        private void beginLooping()
+        {
+            Debug.Assert(!isHandlingLooping);
+
+            music.CurrentTrack.Looping = isHandlingLooping = true;
+
+            music.TrackChanged += ensureTrackLooping;
+        }
+
+        private void endLooping()
+        {
+            Debug.Assert(isHandlingLooping);
```

Comment:
As evidenced by the failing test, `endLooping()` can actually run twice on the same screen if it is suspended, and then exited from due to unwinding the screen stack to a screen lower down. Probably fine to remove this assert, or turn it into a guard statement? Not sure.